.ci/Jenkinsfile

```diff
@@ -56,16 +58,31 @@ pipeline {
       post {
         always {
           archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/eslint-junit.xml")
-          junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/eslint-junit.xml")
+            junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/eslint-junit.xml")
         }
       }
     }
-    stage('Test'){
-      steps {
-        withGithubNotify(context: "Test") {
-          withNodeJSEnv() {
-            dir("${BASE_DIR}"){
-              sh(label: 'npm test', script: 'npm test')
+    stage('Docker test setup') {
+      agent {
+        dockerfile {
+          reuseNode true
+            dir "$BASE_DIR"
+            filename './e2e/Dockerfile.jenkins'
+            additionalBuildArgs  "--build-arg BASE_DIR=\".\" --build-arg DOCKER_BASE_DIR=\"${DOCKER_BASE_DIR}\""
+        }
+      }
```

Comment:
I'm not sure if the `packer_cache.sh` approach works well with this specific declarative approach.

I wonder if this section could be embedded in a shell step itself:

1) Create entrypoint:
```bash
#!/bin/bash
Xvfb ${DISPLAY} -screen 0 1024x768x16 &
npm test
                
source $NVM_DIR/nvm.sh
cd $DOCKER_BASE_DIR
nvm use
npm test
```

2) Create script in the path `.ci/scripts/run-e2e.sh` (or somewhere else)
```bash
docker build --build-arg BASE_DIR=\".\" --build-arg DOCKER_BASE_DIR=\"${DOCKER_BASE_DIR}\" -f ./e2e/Dockerfile.jenkins -t synthetics-recorder:latest .
docker run --rm -ti synthetics-recorder:latest
```

3) Edit pipeline to run

```
stage('Docker test setup') {
  steps {
      withGithubNotify(context: "Test") {
        sh(label: 'run e2e', script: '.ci/scripts/run-e2e.sh')
      }
  }
}
```

The above approach will avoid any dependency with the CI, since what you can run locally in one script can be executed in the CI with one call.